src/core/tree_methods.ts

```diff
@@ -1,11 +1,13 @@
 import * as _ from 'lodash'
 
 export class TreeNode<T> {
-  data: T;
+  name: string;
+  id: string;
   children: TreeNode<T>[];
 
-  constructor(data: T, children: TreeNode<T>[] = []) {
-    this.data = data;
+  constructor(name: string, children: TreeNode<T>[] = []) {
+    this.name = name;
+    this.id = Math.random().toString(36).substr(2, 9);
```

Comment:
0. Please don't do side effects in the constructor => it gets messy quickly and is a pain to test
1. Could you please extract the id generation to a separate function and handle this at node addition level?
2. I find this method quite inelegant, however we should be good for now with those 7 bytes of entropy and string comparison ;)